backend/src/main/java/web/mappers/CertificateMapper.java

```diff
@@ -1,31 +1,91 @@
 package web.mappers;
 
-import data.entities.Certificate;
-import data.entities.Comment;
-import data.entities.User;
+import data.entities.*;
+import data.repositories.SupplierRepo;
+import data.repositories.UserRepo;
+import jakarta.enterprise.context.ApplicationScoped;
+import jakarta.inject.Inject;
 import web.dtos.CertificateDto;
 
 import java.util.List;
 import java.util.Set;
 import java.util.stream.Collectors;
 
+
+@ApplicationScoped
 public class CertificateMapper {
 
+    @Inject
+    SupplierRepo supplierRepo;
+
+    @Inject
+    UserRepo userRepo;
 
-    public static CertificateDto toDto(Certificate certificate){
-        CertificateDto dto= new CertificateDto();
+    public CertificateDto toDto(Certificate certificate) {
+        CertificateDto dto = new CertificateDto();
         dto.setId(certificate.getId());
         dto.setSupplierId(certificate.getSupplier().getId());
         dto.setCertificateType(certificate.getCertificateType().getDisplayName());
         dto.setValidFrom(certificate.getValidFrom());
         dto.setValidTo(certificate.getValidTo());
         dto.setPdfUrl(certificate.getPdfUrl());
-        //no duplicates allowed
-        Set<Long> userIds=certificate.getAssignedUsers().stream().map(User::getId).collect(Collectors.toSet());
+
+        // No duplicates allowed for assigned users
+        Set<Long> userIds = certificate.getAssignedUsers()
+                .stream()
+                .map(User::getId)
+                .collect(Collectors.toSet());
         dto.setAssignedUserIds(userIds);
-        //duplicates can exist( it is just a list)
-        List<String> commentContent=certificate.getComments().stream().map(Comment::getContent).collect(Collectors.toList());
+
+        // Comments can have duplicates (just a list)
+        List<String> commentContent = certificate.getComments()
+                .stream()
+                .map(Comment::getContent)
+                .collect(Collectors.toList());
         dto.setComments(commentContent);
         return dto;
     }
+
+    public Certificate toEntity(CertificateDto certificateDto) {
+        Certificate entity = new Certificate();
+        entity.setId(certificateDto.getId());
+
+        Supplier supplier = supplierRepo.findById(certificateDto.getId());
+        if (supplier != null) {
+            entity.setSupplier(supplier);
+        } else {
+            throw new RuntimeException("Supplier not found");
+        }
+
+        entity.setCertificateType(Certificate_Type.valueOf(certificateDto.getCertificateType()));
```

Comment:
Please use camelCase for naming so CertificateType, also in dto should be enum, no need to parse it from string.